infrastructure/terraform/functions/main.tf

```diff
@@ -11,52 +11,58 @@ resource "azurerm_storage_account" "fxnstor" {
   account_kind             = "StorageV2"
 }
 
-resource "azurerm_app_service_plan" "fxnapp" {
+resource "azurerm_service_plan" "fxnapp" {
   name                = format("%s-%s-fxn-asp-%s", var.appname, var.domainprefix, var.environment)
   location            = var.location
   resource_group_name = var.resource_group_name
-  kind                = "functionapp"
-  sku {
-    tier = "Dynamic"
-    size = "Y1"
-  }
+  os_type             = "Windows"
+  sku_name            = "Y1"
 }
 
-resource "azurerm_function_app" "fxn" {
+resource "azurerm_windows_function_app" "fxn" {
```

Comment:
Why the name "azurerm_windows_function_app" instead of "azurerm_function_app" ? Is it running in Windows container?

Reply:
Because it is deprecated. See https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/function_app